### PR TITLE
Add update actions to integration test

### PR DESCRIPTION
**Pull request by winterqt**
I hope the commit message is descriptive enough -- I went back and forth between a few variations, and I'm not really sure how good this one is.

Additionally, I tried to split up the changes as best as I could, but I can split them up even more if desired.

### CODE REVIEW

**Comment by winterqt** [comment]
cc @ryantm

File: test/integration.nix [hunk 1]
```diff
@@ -1,58 +1,89 @@
...
+          system1.wait_until_succeeds("pgrep login")
+          system1.sleep(2)
+          system1.send_chars("${password}\n")
+          system1.send_chars("whoami > /tmp/1\n")
+          system1.wait_for_file("/tmp/1")
+          assert "${user}" in system1.succeed("cat /tmp/1")
+
+          system2.wait_for_unit("multi-user.target")
+          system2.wait_until_succeeds("${nodes.system2After.config.system.build.toplevel}/bin/switch-to-configuration test")
+          system2.wait_until_succeeds("grep bar /tmp/foo")
```
Comment (Radvendii) [inline test/integration.nix]: Seems like we should also test that this *doesn't* get run at startup. i.e. that `/tmp/foo` doesn't exist until we switch to the new configuration.

**Review by Radvendii — COMMENTED** [review]
We're not testing `services` here. Not sure if that's important to do, since `services` and `action` operate on the same principles.

**Comment by Radvendii** [comment]
Other than that it looks good. Thanks for helping out with this @winterqt.

**Comment by winterqt** [comment]
> We're not testing `services` here. Not sure if that's important to do, since `services` and `action` operate on the same principles.

This was intentional for the reasoning you described, yeah.

**Comment by Radvendii** [comment]
Oh wait, as soon as I merged I discovered an issue. If we run it through `nix flake check`... well first of all we have to change `pkgs` to `nixpkgs` in the arguments to `integration.nix`. But more importantly it's complaining about 

```
would reformat /nix/store/g86ip3j1mccbffhr86ya6l1arrncp3r0-nixos-test-driver-agenix-integration/test-script
Oh no! 💥 💔 💥
1 file would be reformatted.
```

**Comment by Radvendii** [comment]
fixed it; it was a linker error. For the record I'm grumpy about linkers that will complain about a line of code that's longer than 80 characters, and then suggest splitting it into 3 lines of code, only one of which is longer than 80 characters.

**Pull request merged**